main.go

```diff
@@ -86,12 +89,22 @@ func main() {
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
+	mux := http.NewServeMux()
```

Comment:
Have you considered using this https://github.com/kubernetes-sigs/controller-runtime/pull/1943/files ?